checker/src/main/java/org/checkerframework/checker/pico/jdk.astub

```diff
@@ -0,0 +1,355 @@
+import org.checkerframework.checker.pico.qual.Mutable;
+import org.checkerframework.checker.pico.qual.Immutable;
+import org.checkerframework.checker.pico.qual.ReceiverDependentMutable;
+import org.checkerframework.checker.pico.qual.Readonly;
+import org.checkerframework.checker.pico.qual.ObjectIdentityMethod;
+import java.util.Collection;
+
+package java.lang;
+
+@ReceiverDependentMutable
+class Iterable<E> {
+  Iterator<E> iterator(@Readonly Iterable this);
```

Comment:
```suggestion
  @PolyMutable Iterator<E> iterator(@PolyMutable Iterable this);
```

Should this use `PolyMutable`? We should make sure that for a readonly/immutable Iterable, we cannot produce a mutable Iterator, which the current signature incorrectly allows.
By using PolyMutable, we connect the mutability of the two.